Automattic/woocommerce-payments
# Track downloads on deposit details page.

Fixes #5885 
#### Changes proposed in this Pull Request

![image](https://github.com/Automattic/woocommerce-payments/assets/57298/ff8cec8e-821a-40a3-b55c-06756f64709c)

This PR updates tracking on the deposit details page to add clarity to event tracking of downloads.

Previously downloads from this page have the ambiguous event `wcadmin_wcpay_transactions_download` this is due to the transactions and deposit details page both using the same component with the only difference being a `depositId`. ~This PR updates the event when downloading a deposits transactions to use `wcpay_deposits_detail_download`~

~This change will break any tracking flows for anything that is currently setup as it is currently assumed that deposit details downloads will have the `wcadmin_wcpay_transactions_download` event firing. However, there do not appear to be any current flows/funnels setup for this event. A note should be added on this event to make sure that viewers are aware.~

This PR now adds a new event `wcpay_transactions_download_csv_click` which fires when the button is clicked. The previous event `wcpay_deposits_detail_download` only fires when the download is successful. The new event contains the following attributes

 * `location`: `transactions` or `deposit_details` to signify where the button was clicked
 * `download_type`: `endpoint` or `browser` 
 * `exported_transactions`: Amount of transactions included in the downloaded report.
 * `total_transactions`: Total amount of transactions available with the used filters.




#### Testing instructions

* Visit the WC Pay transactions page and confirm that the `wcpay_transactions_download_csv_click` still fires when the "Download CSV" button is clicked.
* Visit the deposits page, click through to a deposit, confirm that the `wcpay_transactions_download_csv_click` still fires when the "Download CSV" button is clicked.

-------------------


**Post merge**
